asyncio_mqtt/client.py

```diff
@@ -250,6 +250,7 @@ def __init__(  # noqa: C901
         will: Will | None = None,
         clean_session: bool | None = None,
         transport: str = "tcp",
+        timeout: int | float = 10,
```

Comment:
I propose that we use `timeout: float | None = None`. Note that `int` is redundant as per PEP 484:

> [...] when an argument is annotated as having type float, an argument of type int is acceptable; [...]

Thanks to @JonathanPlasse for pointing that out. +1

Note that we use `None` to mean "Give me the default value". It doesn't mean "no timeout". If you want "no timeout", use `math.inf` instead. :)